File: apps/js-sdk/firecrawl/src/v2/utils/agentHints.ts
/** Keep response guidance as metadata when convenience methods unwrap data. */
export function agentHintMetadata(body: unknown): { agent_hints?: string[] } {
  const hints = (body as { agent_hints?: unknown } | null)?.agent_hints;
  return Array.isArray(hints) && hints.every((hint) => typeof hint === "string")
    ? { agent_hints: [...hints] }
    : {};
}

File: apps/python-sdk/firecrawl/v2/utils/agent_hints.py
"""Preserve response guidance separately from extracted document content."""
from typing import Any, Dict, List


def agent_hint_metadata(body: Any) -> Dict[str, List[str]]:
    hints = body.get("agent_hints") if isinstance(body, dict) else None
    if isinstance(hints, list) and all(isinstance(hint, str) for hint in hints):
        return {"agent_hints": list(hints)}
    return {}
